modules/repository-url/src/test/java/org/elasticsearch/common/blobstore/url/URLBlobContainerRetriesTests.java

```diff
@@ -91,7 +89,7 @@ protected BlobContainer createBlobContainer(Integer maxRetries,
             final URLHttpClientSettings httpClientSettings = URLHttpClientSettings.fromSettings(settings);
             URLBlobStore urlBlobStore =
                 new URLBlobStore(settings, new URL(getEndpointForServer()), factory.create(httpClientSettings), httpClientSettings);
-            return urlBlobStore.blobContainer(path == null ? BlobPath.EMPTY : path);
+            return urlBlobStore.blobContainer(BlobPath.EMPTY);
```

Comment:
No randomisation here for now since this thing doesn't raise consistent exceptions with S3 and GCS (doesn't add the relative path the same way) and I figured with the URL repository potentially on the way out it's not worth the effort to align the behavior right now (it's not 100% straight forward because of string magic in building relative urls there).